salt/modules/incron.py

```diff
@@ -103,7 +105,7 @@ def _write_incron_lines(user, lines):
     else:
         path = salt.utils.files.mkstemp()
         with salt.utils.files.fopen(path, 'w+') as fp_:
-            fp_.writelines(lines)
+            fp_.writelines(salt.utils.stringutils.to_str(lines))
```

Comment:
`to_str()` has to operate on strings, not lists. If you want a one-liner that doesn't use a list comprehension, you can open the file using a mode of `wb` and use:

```python
fp_.writelines(salt.utils.data.encode(lines))
```

Just FYI, the `+` in the mode is entirely unnecessary, it opens the file in read-and-write mode and we're not reading from the filehandle at all. It was likely originally written that way by someone who didn't understand what the `+` was for.